pkg/k8s/watchers/pod.go

```diff
@@ -639,26 +670,53 @@ func (k *K8sWatcher) deleteHostPortMapping(pod *slim_corev1.Pod, podIPs []string
 	return nil
 }
 
-func (k *K8sWatcher) updatePodHostData(pod *slim_corev1.Pod, podIPs []string) error {
-	err := k.upsertHostPortMapping(pod, podIPs)
+func (k *K8sWatcher) updatePodHostData(oldPod, newPod *slim_corev1.Pod, oldPodIPs, newPodIPs []string) error {
+	var namedPortsChanged bool
+	defer func() {
+		// delete all IPs that were not added regardless if the insertion of the
+		// entry in the ipcache map was successful or not because we will not
+		// receive any other event with these old IP addresses.
+		for _, oldPodIP := range oldPodIPs {
+			var found bool
+			for _, newPodIP := range newPodIPs {
+				if newPodIP == oldPodIP {
+					found = true
+					break
+				}
+			}
+			if !found {
+				npc := ipcache.IPIdentityCache.Delete(oldPodIP, source.Kubernetes)
+				if npc {
+					namedPortsChanged = npc
+				}
```

Comment:
```suggestion
				npc := ipcache.IPIdentityCache.Delete(oldPodIP, source.Kubernetes)
				if npc {
					namedPortsChanged = true
				}
```

I think this makes it clearer because at first I thought that the code was overwriting `namedPortsChanged` to the last value of `npc` which is not what it's doing.